Reminder-App/Reminder-App/ReminderListViewController.swift

```diff
@@ -9,32 +9,217 @@ import UIKit
+    private var filter: ReminderListDataSource.Filter {
+        return ReminderListDataSource.Filter(rawValue: filterSegmentedControl.selectedSegmentIndex) ?? .today
+    }
     
-    // MARK: Static Properties
+    // MARK: Public Properties
     static let showDetailSegueIdentifier = "ShowReminderDetailSegue"
     static let reminderListCellIdentifier = "ReminderListCell"
+    static let mainStoryboardName = "Main"
+    static let detailViewControllerIdentifier = "ReminderDetailViewController"
     
     // MARK: Overridden Methods
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        
         if segue.identifier == Self.showDetailSegueIdentifier,
            let destination = segue.destination as? ReminderDetailViewController,
            let cell = sender as? UITableViewCell,
+           
            let indexPath = tableView.indexPath(for: cell) {
+            let rowIndex = indexPath.row
+            
+            guard let reminder = reminderListDataSource?.reminder(at: rowIndex) else {
+                return
+            }
             
-            let reminder = Reminder.testData[indexPath.row]
-            destination.configure(with: reminder)
+            destination.configure(with: reminder, editAction: {
+                reminder in
+                self.reminderListDataSource?.update(reminder, at: rowIndex) {
+                    success in
+                    
+                    if success {
+                        DispatchQueue.main.async {
+                            self.tableView.reloadData()
+                            self.refreshProgressView()
+                        }
+                    } else {
+                        DispatchQueue.main.async {
+                            let alertTitle = NSLocalizedString(ErrorMessages.cannotUpdateReminderTitle, comment: ErrorMessages.errorUpdatingReminderTitle)
+                            let alertMessage = NSLocalizedString(ErrorMessages.errorInUpdatingReminder, comment: ErrorMessages.errorUpdatingReminderMessage)
+                            
+                            let alert = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+                            let actionTitle = NSLocalizedString(ErrorMessages.okMessage, comment: ErrorMessages.okActionTitle)
+                            
+                            alert.addAction(UIAlertAction(title: actionTitle,
+                                                          style: .default,
+                                                          handler: {
+                                                            _ in
+                                                            self.dismiss(animated: true, completion: nil)
+                                                          }))
+                            
+                            self.present(alert, animated: true, completion: nil)
+                        }
+                    }
```

Comment:
you can convert an `if else` statement into a `guard` statement such that the shorter piece of code should be inside the `else` block of `guard`. and the longer chunk of code should come after the `guard` statement.